src/Orleans/Streams/Internal/StreamSubscriptionHandleImpl.cs

```diff
@@ -79,14 +86,16 @@ public override Task<StreamSubscriptionHandle<T>> ResumeAsync(IAsyncObserver<T>
                 await NextItem(itemTuple.Item1, itemTuple.Item2);
             }
 
-            expectedToken = StreamHandshakeToken.CreateDeliveyToken(batch.SequenceToken);
-
+            if (isRewindable)
+            {
+                expectedToken = StreamHandshakeToken.CreateDeliveyToken(batch.SequenceToken);
+            }
             return null;
         }
 
         public async Task<StreamHandshakeToken> DeliverItem(object item, StreamSequenceToken currentToken, StreamHandshakeToken handshakeToken)
         {
-            if (expectedToken != null)
+            if (isRewindable && expectedToken != null)
```

Comment:
You are correct.  Since the expected token is never set, it will be null, and the delivery should function as expected.
This is a nice fix.  Much simpler than I would have suspected necessary :)

Now that I understand your approach, I don't believe the isRewindable checks are necessary prior to the token check at the start of each delivery function.  It is only necessary to check isRewindable prior to setting the expected token.  Since the existing token check only occurs if an expected token is set, they will work correctly, without the isRewindable check if we never set the expected token.

The heart of this change is really about never setting the expected token.  If the expected token is never set, no ordering corrections occur and new agents always deliver data from the start of the received data on a stream.